### ISSUE TITLE
COUNT_VARARGS

### ISSUE THREAD

**Issue by Islam0mar** [elcritch/forthwith#2]
```c
#define _GET_NTH_ARG(_1, _2, _3, _4, _5, _6, _7, _8, _9, _10, _11, _12, _13, _14, _15, _16, _17, _18, _19, _20, _21, N, ...) N
#define COUNT_VARARGS(...) _GET_NTH_ARG(__VA_ARGS__, 20, 19, 18, 17, 16, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1)
```
Looking through your code, I found [here](https://github.com/elcritch/forthwith/blob/master/src/xmacros.h#L13) _21 ~ 20, then you add 1 or 2 to  COUNT_VARARGS(...)
Why are you using it this way?
Shouldn't the numbering be identical? as in [there](https://codecraft.co/2014/11/25/variadic-macros-tricks/)

BTW:
The part of forth macro definitions is amazing. I am copying your style :+1:

**Comment by elcritch** [comment]
Likely a bug :-) I think I got the arg-count trick from that site. There's a small chance those two macros need to be offset for some reason... Feel free to submit a PR if you like.  

> The part of forth macro definitions is amazing. I am copying your style 👍

Thanks! It was a fun hack. Every other Forth appeared to use GCC linker scripts which didn't work on the platform I was targeting. Turns out xmacro's work pretty well. The ARM port only took a few hours.
